src/utils/popups.js

```diff
@@ -55,28 +58,70 @@ const wcLinkPopupHTML = (title, displayProps) => {
           ${downstreamWCLinks}`;
 };
 
+const fitMapToFeatures = (map, features) => {
+  var bounds = new mapboxgl.LngLatBounds();
+
+  features.forEach((feature) => {
+    if (feature.geometry.type === "LineString") {
+      feature.geometry.coordinates.forEach((coords) => {
+        bounds.extend(coords);
+      });
+    } else if (feature.geometry.type === "MultiLineString") {
+      feature.geometry.coordinates.forEach((line) => {
+        line.forEach((coords) => {
+          bounds.extend(coords);
+        });
+      });
+    }
+  });
+
+  map.fitBounds(bounds, { padding: 50 });
+};
+
 const getUpstream = async (id) => {
-  const url = `${waterNetworkAPIBase}/collections/WatercourseLink/items/${id}/upstream`;
+  const url = `${waterNetworkAPIBase}/collections/WatercourseLink/items/${id}/upstream?limit=10`;
   return await getURL(url);
 };
 // TODO: catch 404s
 const highlightUpstreamWatercourseLinks = async (id, map) => {
-  await getUpstream(id).then((watercourseLinks) => {
-    closePopup();
-    map.getSource("highlightWatercourseLink").setData(watercourseLinks);
-  });
+  const upstreamWcLinks = await getUpstream(id);
+  clearUpstreamDownstream(map);
```

Comment:
Going for a _"simple"_ approach first, where the user can only look at upstream or downstream at a time

not sure if this is what's ideal for the "user story" but thought this would also be a good way to keep the bbox small and would mean we don't have to worry about fitting both upstream and downstream wclinks in the viewport (if that turns out to ne not necessary!)